jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -215,6 +222,43 @@ void completeInitialization() {
 	/*[IF Sidecar19-SE|Sidecar18-SE-OpenJ9]*/
 	System.startSNMPAgent();
 	/*[ENDIF]*/ // Sidecar19-SE|Sidecar18-SE-OpenJ9
+
+	/*[IF JAVA_SPEC_VERSION >= 11] */
+	// Now that all modules and Charset's are available, update System.out and err if necessary
+	Properties props = System.internalGetProperties();
+	Charset defaultCharset = Charset.defaultCharset();
+	String fileEncoding = props.getProperty("file.encoding"); //$NON-NLS-1$
+
+	// If the sun.stderr.encoding was already set in System, don't set it again
+	String stderrEncoding = System.hasSetErrEncoding ? null : props.getProperty("sun.stderr.encoding"); //$NON-NLS-1$
+	String stderrCharset = System.getCharsetName(stderrEncoding, fileEncoding, defaultCharset);
+	if (stderrCharset != null) {
+		try {
+			System.err.flush();
+			/*[IF PLATFORM-mz31|PLATFORM-mz64]*/
+			System.setErr(com.ibm.jvm.io.ConsolePrintStream.localize(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true, stderrCharset));
+			/*[ELSE]*/
+			System.setErr(new PrintStream(new BufferedOutputStream(new FileOutputStream(FileDescriptor.err)), true, stderrCharset));
+			/*[ENDIF] PLATFORM-mz31|PLATFORM-mz64 */
+		} catch (UnsupportedEncodingException e) {
+		}
+	}
+
+	// If the sun.stdout.encoding was already set in System, don't set it again
+	String stdoutEncoding = System.hasSetOutEncoding ? null : System.internalGetProperties().getProperty("sun.stdout.encoding"); //$NON-NLS-1$
```

Comment:
This should use `props` instead of calling `System.internalGetProperties()` again.